zos/setupEcosystem.sh

```diff
@@ -223,75 +286,140 @@ if checkForAndInstall $PYTHON_DIR; then
+        echo "Python Install: Fail:" >> $STATUS_LOG
+        exit 103
+    else
+          echo "python installed successfully" >> $PYTHON_LOG
+          echo "Python Install: Success" >> $STATUS_LOG
 
-   if [ $? != 0 ]; then
-    echo "Python Installation failed"
-    exit 103
+    fi
+  else
+    echo "Python Installation failed because it has not been succesfully unpackaged" >> $PYTHON_LOG
+    echo "Python Install: Fail" >> $STATUS_LOG
   fi
+    echo "python installation done"
 
-  echo "python installation done"
 fi
-
 setupPythonVariables
 
+if [ "$(tail -1 $STATUS_LOG)" != "Python Install: Success" ]; then
+echo "Cannot Begin Torre install as their was an issue with the python Installation"
+echo "$(tail -1 $STATUS_LOG)"
+exit 103
+fi
 # Toree Install
 if checkForAndInstall $TOREE_DIR; then
-  echo "Installing Toree"
+  echo "Installing Toree" >>  $STATUS_LOG
   cd $TOREE_DIR
+  echo "Starting upackaging of Apache Toree" >> $TOREE_LOG
+  #decompress torre and check return code
   cat $TAR_DIR/$TOREE_TAR$TAR_EXTENSION |  gunzip -c | tar xUXof -
-
    if [ $? != 0  ]; then
-    echo "Torre Unpackaging failed"
+    echo "Torre Unpackaging failed: returned a non zero return code" >> $TOREE_LOG
+     echo "Torre Unpackaging: Fail" >> $STATUS_LOG
     exit 111
-  fi
+   fi
 
   cd $TOREE_TAR
   chtag -t -c iso8859-1 -R *
-  # TODO: add checks to build/install
-  python setup.py build
+  # builds and checks return code
+  python setup.py build &>> $TOREE_LOG
    if [ $? != 0 ]; then
-    echo "Build failure"
+    echo "Toree Build Failed: returned a non zero return code" >> $TOREE_LOG
     exit 112
   fi
-  python setup.py install
+  #install torre
+  python setup.py install &>> $TOREE_LOG
    if [ $? != 0 ]; then
-    echo "Installation Failure"
+    echo "Toree Installation Failure: returned a non zero return code" >> $TOREE_LOG
+    echo "Torre Install: Fail" >> $STATUS_LOG
     exit 113
+    else
+    echo "Toree installed successfully" >> $TOREE_LOG
+    echo "Toree Installation: Success" >> $STATUS_LOG
   fi
-  convertToEBCDIC $PYTHON_EXTRACTED/$PYTHON_VERSION/bin/jupyter
-  jupyter toree install --user
+  # check log to make sure toree install was completed succesfully
+   if [ "$(tail -1 $STATUS_LOG)" = "Toree Installation: Success" ]; then
+   python $PYTHONHOME/bin/jupyter toree install --user &>> $TOREE_LOG
+    if [ $? != 0 ]; then
+         echo "user Installation Failure" >> $TOREE_LOG
+         exit 113
+    else echo "Toree install: success" >> $TOREE_LOG
+         echo "Toree User Installation: Success" >> $STATUS_LOG
+    fi
+   else
+    echo "user Installation Failure" >> $TOREE_LOG
+    echo "Toree User Installation: Fail" >> $STATUS_LOG
+   fi
   convertToEBCDIC $HOME/.local/share/jupyter/kernels/apache_toree_scala/bin/run.sh
 fi
+ #check log to make sure Install completed succesfully
+ if [ "$(tail -1 $STATUS_LOG)" = "Toree Installation: Success" ]; then
```

Comment:
Doesn't appear to be same message, use constants so you only have to change in one place